arch/Kconfig

```diff
@@ -72,6 +72,32 @@ config BIG_ENDIAN
          is used to select linker script OUTPUT_FORMAT and command
          line option for gen_isr_tables.py.
 
+config TRUSTED_EXECUTION_SECURE
+	bool "Trusted Execution: Secure firmware image"
+	depends on ARCH_HAS_TRUSTED_EXECUTION
+	help
+	  Select this option to enable building a Secure firmware
+	  image for a platform that supports Trusted Execution. A
+	  Secure firmware image will execute in Secure state. It may
+	  allow the CPU to execute in Non-Secure (Normal) state.
+	  Therefore, a Secure firmware image shall be able to
+	  configure security attributions of CPU resources (memory
+	  areas, peripherals, interrupts, etc.) as well as to handle
+	  faults, related to security violations. It may optionally
+	  allow certain functions to be called from the Non-Secure
+	  (Normal) domain.
+
+config TRUSTED_EXECUTION_NONSECURE
+	bool "Trusted Execution: Non-Secure firmware image"
+	depends on ARCH_HAS_TRUSTED_EXECUTION
+	help
+	  Select this option to enable building a Non-Secure
+	  firmware image for a platform that supports Trusted
+	  Execution. A Non-Secure firmware image will execute
+	  in Non-Secure (Normal) state. Therefore, it shall not
+	  access CPU resources (memory areas, peripherals,
+	  interrupts etc.) belonging to the Secure domain.
```

Comment:
Could have an `if` here:

```
if ARCH_HAS_TRUSTED_EXECUTION

config TRUSTED_EXECUTION_SECURE
	...

config TRUSTED_EXECUTION_NONSECURE
	...

endif # ARCH_HAS_TRUSTED_EXECUTION
```

Reply:
We could do that @ulfalizer , but we don't do it for other stuff below, e.g. HW_STACK_PROTECTION and USERSPACE. So I wonder, if it is better to keep consistency among all these user option. If you insist, I am ok to add the "if" guards

Reply:
Think it sets a nice precedent, though it's not a huge deal when it's just two symbols.

Maybe the other ones could be changed in a separate PR, or all of them in a separate PR, if you're going for consistency. Bit silly to block in that case. :)

Wonder if Kconfig would be nicer if the "standard" style indented blocks. Would look a bit off when you're used to unindented though...

Reply:
Added the if guards.

Reply:
I believe TRUSTED_EXECUTION_SECURE and TRUSTED_EXECUTION_NONSECURE are mutual exclusive: a configuration will either produce a secure binary image, or a normal one and not both. Add a "depends on !TRUSTED_EXECUTION_SECURE" or so to ensure both options aren't selected at the same time?

Reply:
This is good suggestion, @ruuddw , i will add it

Reply:
We have this mutual exclusiveness in the internal (arm) options, but, it is good idea to add it also in the high level Kconfig options.